storage/watchable_store.go

```diff
@@ -185,7 +185,12 @@ func (s *watchableStore) watch(key []byte, prefix bool, startRev int64, ch chan<
 
 	k := string(key)
 	if startRev == 0 {
-		s.synced[k] = append(s.synced[k], wa)
+		if v, ok := s.synced[k]; ok {
+			v[wa] = struct{}{}
```

Comment:
let's check if v[wa] exists. It should not exist before or it means we put in the same watch twice. Probably we should just panic if it exists. But please double check.